### PR TITLE
Port Graph to OpenGL

### PR DESCRIPTION
**Pull request by eulertour**
## Changelog / Overview

Port Graph to OpenGL

Depends on #1200 

## Motivation

Allows using Graph in OpenGL

## Explanation for Changes

Ports LabeledDot and Graph to OpenGL

## Testing Status

None, though I'd be willing to add some if I can find how to create OpenGL test data.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/animation/transform.py
Comment (jsonvillanueva) [inline manim/animation/transform.py:84]:
```suggestion
        if config.renderer == "opengl":
```

File: manim/mobject/mobject.py
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:209]:
```suggestion
        if config.renderer == "opengl":
```

File: manim/utils/space_ops.py
Comment (jsonvillanueva) [inline manim/utils/space_ops.py:235]:
```suggestion
    if config.renderer == "opengl":
```

File: manim/mobject/opengl_geometry.py
Comment (jsonvillanueva) [inline manim/mobject/opengl_geometry.py:356]:
```suggestion
    """A :class:`~.OpenGLDot` containing a label in its center.
```

File: manim/mobject/svg/svg_path.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_path.py:93]:
```suggestion
        if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_path.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_path.py:110]:
```suggestion
        if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_path.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_path.py:139]:
```suggestion
            if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_path.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_path.py:176]:
```suggestion
            if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_mobject.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_mobject.py:410]:
```suggestion
            if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_mobject.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_mobject.py:209]:
```suggestion
            if config.renderer == "opengl":
```

File: manim/mobject/svg/svg_mobject.py
Comment (jsonvillanueva) [inline manim/mobject/svg/svg_mobject.py:204]:
```suggestion
        if config.renderer == "opengl":
```

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]
So far this looks good. `config.renderer == "opengl"` is more recent/not deprecated, so just minor issues. I'm still looking through the three largest files (opengl_graph.py, opengl_tex_mobject.py, and opengl_text_mobject.py). More on those files later, but I have to run for now.

**Comment by behackl** [comment]
The actual diff for this PR should be much smaller (and as soon as the merge conflict is resolved it should show in the GH interface), given that the #1200 has already been merged. I've diffed `graph.py` and `opengl_graph.py`, the changes there pretty much look like all of this could be covered by the metaclass approach discussed in #1272 without any further adjustments.

Given that I don't have a lot of time in the coming days, I fear that #1272 won't be finalized/merged too soon -- so in principle I'd be fine with merging this PR and adding `opengl_graph.py` to the list of files that will be removed again after #1272 allows avoiding code duplication.

**Comment by jsonvillanueva** [comment]
> The actual diff for this PR should be much smaller (and as soon as the merge conflict is resolved it should show in the GH interface), given that the #1200 has already been merged.

@eulertour Since that's the case, I'll wait until the merge-conflicts are handled before finishing my review. Just request a review or ping me on Discord and I'll get to it ASAP.

**Comment by behackl** [comment]
I'd prefer to make `Graph` available via #1512 and avoid duplication of the `graph` module.

**Comment by behackl** [comment]
Closing this, as #1512 has been merged.

**Pull request closed without merging**
